### DISCUSSION TITLE
Linkerd's latest stable chart is not being picked up

### DISCUSSION THREAD

**Discussion by alpeb** [artifacthub/hub#1961] [Q&A] [answered]
Hi, we've released a new version (2.11.2) a couple of weeks ago to https://helm.linkerd.io/stable, but the site is still showing up the old one:
https://artifacthub.io/packages/helm/linkerd2/linkerd2
I've updated today the index.yaml trying to trigger the update but that didn't do the trick
![image](https://user-images.githubusercontent.com/554287/167701439-3e9cd5f8-e63e-45e5-8705-9f3ce1b30d89.png)
Thanks in advance :-)

**Comment by tegioz** [comment]
Hi @alpeb 👋 

I've just forced the `linkerd2` repo to be indexed again and the new version has just been published. Updating the `index.yaml` file should have triggered a reindex as well, unless you modified the `generated` field, which we ignore when computing the index hash.

Before forcing the reindex I've checked the database and the logs. The `index.yaml` file hash stored corresponds to the current `index.yaml` and it was processed on `2022-04-21T19:30:27`. But when that happened, no new packages were detected by the tracking job (surprisingly). I remember [something similar happened once a few months ago](https://cloud-native.slack.com/archives/C0103CTR3RB/p1642245262007800?thread_ts=1642206529.007600&cid=C0103CTR3RB), maybe it's the same problem (excerpt from that conversation):

> Once we detect that a repository has been updated, we trigger a job to index the changes. This job fetches the index.yaml file again. Surprisingly, no new packages were detected by the job. The only reason I can think of right now that may have led to this is that somehow the second request was served by a different server that still had the previous index.yaml version cached. 

I'll look into it a bit more tomorrow morning 🙂 

Thanks for reporting this!

**Comment by alpeb** [comment]
Thanks for the prompt reply @tegioz, and for having forced the refresh!
Indeed, since there were no chart changes, my update consisted on just changing the `generated` field :slightly_smiling_face: 
But no idea still what caused change to be ignored in the first place, a couple of weeks ago...

**Comment by tegioz** [comment, marked as answer]
No worries!

My best guess is that somehow we received a stale copy on the second request to get the repo's `index.yaml` file.

Anyway, we should be able to handle gracefully such cases. So we've just added a check for this potential mismatch (#1965). If it happens again we should detect and report it. In those cases, we won't update the repository's digest in the database either so that the repository is processed again in the next tracker run (~30 minutes).

We'll deploy this later today, please let us know if it happens again 🙂

**Reply by alpeb** [reply]
Awesome, thanks again @tegioz !
